lib/shelley/test/unit/Cardano/Wallet/Shelley/TransactionSpec.hs

```diff
@@ -3281,6 +3281,10 @@ prop_balanceTransactionValid wallet (ShowBuildable partialTx') seed
                 label "outputs below minCoinValue" $ property True
             Left (ErrBalanceTxExistingCollateral) ->
                 label "existing collateral" True
+            Left (ErrBalanceTxExistingTotalCollateral) ->
+                label "existing total collateral" True
```

Comment:
@Unisay I think these labels are great, thanks for adding them.

However, as you probably know, using `label` within the body of a pattern match doesn't actually cause QuickCheck to verify that this particular case is covered during a test run.

For that, we'd need to use some combination of `checkCoverage` and `cover` _outside_ of the pattern match. See https://input-output.atlassian.net/browse/ADP-1843

It shouldn't block this PR from being merged (because this PR is a strict improvement over `master`), but I do think we should add coverage checks in a future PR.

Reply:
I completely agree that coverage checks should be implemented. I also agree that its a good idea not to block this PR. https://input-output.atlassian.net/browse/ADP-1843 seems like a proper task for it.